File: Subsets/Subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        output = []

        def dfs(nums, path):
            output.append(path)
            if not nums:
                return
            else:
                for i in range(len(nums)):
                    dfs(nums[i+1:], path + [nums[i]])

        dfs(nums, [])
        return output

File: Subsets/Subsets.rs
impl Solution {
    pub fn subsets(nums: Vec<i32>) -> Vec<Vec<i32>> {
        let mut output: Vec<Vec<i32>> = vec![];
        fn dfs(nums: &Vec<i32>, path: Vec<i32>, output: &mut Vec<Vec<i32>>) {
            output.push(path.to_vec());
            if nums.len() == 0 {
                return;
            } else {
                for i in 0..nums.len() {
                    let mut path_copy = path.clone();
                    path_copy.push(nums[i]);
                    dfs(&nums[i + 1..].to_vec(), path_copy, output);
                }
            }
        }
        let mut path: Vec<i32> = vec![];
        dfs(&nums, path, &mut output);
        return output;
    }
}
